main/src/com/google/refine/exporters/sql/SqlExporter.java

```diff
@@ -0,0 +1,161 @@
+ * 
+ * Neither the name of Google nor the names of its contributors may be used to 
+ * endorse or promote products derived from this software without specific 
+ * prior written permission.
+ * 
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" 
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, 
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR 
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR 
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, 
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, 
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; 
+ * OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, 
+ * WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR 
+ * OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF 
+ * ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package com.google.refine.exporters.sql;
+
+import java.io.IOException;
+import java.io.Writer;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Properties;
+
+import org.json.JSONObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.refine.ProjectManager;
+import com.google.refine.browsing.Engine;
+import com.google.refine.exporters.CustomizableTabularExporterUtilities;
+import com.google.refine.exporters.TabularSerializer;
+import com.google.refine.exporters.WriterExporter;
+import com.google.refine.model.Project;
+import com.google.refine.util.JSONUtilities;
+
+public class SqlExporter implements WriterExporter {
+    
+    public static final String NO_COL_SELECTED_ERROR = "****NO COLUMNS SELECTED****";
+    public static final String NO_OPTIONS_PRESENT_ERROR = "****NO OPTIONS PRESENT****";
+
+    private final static Logger logger = LoggerFactory.getLogger("SqlExporter");
+
+    private List<String> columnNames = new ArrayList<String>();
+    private List<ArrayList<SqlData>> sqlDataList = new ArrayList<ArrayList<SqlData>>();
+    private JSONObject sqlOptions;
+
+    @Override
+    public String getContentType() {
+        return "text/plain";
+    }
+
+    @Override
+    public void export(final Project project, Properties params, Engine engine, final Writer writer)
+            throws IOException {
+        if(logger.isDebugEnabled()) {
+            logger.debug("export sql with params: {}", params);
+        }
+        TabularSerializer serializer = new TabularSerializer() {
+
+            @Override
+            public void startFile(JSONObject options) {
+                sqlOptions = options;
+                //logger.info("setting options::{}", sqlOptions);
+            }
+
+            @Override
+            public void endFile() {
+                try {
+                    if (columnNames.isEmpty()) {
+                        writer.write(NO_COL_SELECTED_ERROR);
```

Comment:
Should this be controlled from the front end? We can have the validation at back end, but we may not need write the error into the dump file.

Reply:
There is front end validation as well if you try to generate sql with no column selected. I can suppress the error dump but if this happens, the user will see an empty file.